subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileToolchainIntegrationTest.groovy

```diff
@@ -337,11 +339,50 @@ public class Foo {
         outputContains("task.targetCompatibility = $targetOut")
 
         where:
-        source  | target    | sourceOut | targetOut
-        '9'     | '10'      | '1.9'     | '1.10'
-        '9'     | 'none'    | '1.9'     | '11'
-        'none'  | 'none'    | '11'      | '11'
+        source | target | sourceOut | targetOut
+        '9'    | '10'   | '1.9'     | '1.10'
+        '9'    | 'none' | '1.9'     | '11'
+        'none' | 'none' | '11'      | '11'
+
+    }
+
+    def "can compile Java using different JDKs"() {
```

Comment:
Maybe use `#javaVersion` to see which iterations are skipped?

Reply:
We get this information without explicitly embedding the variable in the test header (since Spock 2 I believe): https://ge.gradle.org/s/vhyeam6vqgtqi/tests
![image](https://user-images.githubusercontent.com/5560673/107525745-d1efa080-6bbf-11eb-8bb9-6cc3fce098a6.png)